SDAPackage/SDA/obstacle_map.py

```diff
@@ -27,11 +27,56 @@ def add_obstacle(self, obstacle_to_add):
         :param obstacle_to_add: The obstacle to add to the map
         :type obstacle_to_add: StationaryObstacle
         """
+        final_obstacles = np.array([])
         if self.obstacles.size != 0:
-            self.obstacles = np.hstack((self.obstacles, obstacle_to_add))
+            final_obstacles = np.hstack((self.obstacles, obstacle_to_add))
+            final_obstacles = overlapping_obstacles(final_obstacles)
+            self.obstacles = np.hstack((self.obstacles, final_obstacles))
         else:
             self.obstacles = np.array([obstacle_to_add])
 
+    def overlapping_obstacles(self, final_obstacles):
+        for first_obstacle_index in range(len(final_obstacles)):
+            for second_obstacle_index in range(first_obstacle_index + 1, len(final_obstacles)):
+                if (overlap_obstancles_boolean(first_obstacle_index, second_obstacle_index, final_obstacles) is False)
+                    obstacle_for_adding = make_encompassing_circle(first_obstacle_index, second_obstacle_index, final_obstacles)
+                    np.delete(final_obstacles, first_obstacle_index)
+                    np.delete(final_obstacles, second_obstacle_index)
+                    np.hstack(final_obstacles, obstacle_for_adding)
```

Comment:
I don't believe this function will execute correctly. I think you want a call like this: ```final_obstacles = np.hstack([final_obstacles, obstacle_for_adding])```

Reply:
done